src/applications/apps.ts

```diff
@@ -383,75 +411,95 @@ function validCustomProps(customProps) {
   );
 }
 
-function sanitizeArguments(
-  appNameOrConfig,
-  appOrLoadApp,
-  activeWhen,
-  customProps
-) {
+interface ApplicationRegistration {
+  name: string;
+  loadApp: Application;
+  activeWhen: ActivityFn;
+  customProps?: CustomProps;
+}
+
+function sanitizeArguments<ExtraProps extends CustomProps = {}>(
+  appNameOrConfig: string | RegisterApplicationConfig,
+  appOrLoadApp: Application,
+  activeWhen: Activity,
+  customProps?: ExtraProps | CustomPropsFn<ExtraProps>
+): ApplicationRegistration {
   const usingObjectAPI = typeof appNameOrConfig === "object";
 
-  const registration = {
-    name: null,
```

Comment:
To facilitate stricter compile-time type checks, I made a runtime change where the object is initialized with all necessary properties rather than starting off with null properties.